### PR TITLE
Changed theme responsive breakpoints

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed theme responsive breakpoints.
Themes can integrate customized breakpoints more fully.
Removed `narrow` and `wide` responsive labels and replaced them with `small`, `medium`, and `large`.
Refactored how breakpoint customizations are stored in the theme. For example, we used to have `theme.global.edgeSize.narrow`. That has been replaced by `theme.global.breakpoints.small.edgeSize`.

#### Where should the reviewer start?

base.js

#### What testing has been done on this PR?

storybook

#### How should this be manually tested?

storybook
review snapshot changes

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/2334
https://github.com/grommet/grommet/issues/2285
https://github.com/grommet/grommet/issues/2284

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

definitely

#### Is this change backwards compatible or is it a breaking change?

If you use `ResponsiveContext`, this is definitely a breaking change. Sorry about that.
If you have a custom theme, this may be a breaking change.

### CODE REVIEW

**Comment by alansouzati** [comment]
Can we be more specific on the theme changes to help developers understand fully what they have to change to resolve the breaking changes?

**Comment by alansouzati** [comment]
We need to update the `custom breakpoints` example. this pr breaks that. it would be good also because it will serve as a reference to folks wanting to leverage this.

**Comment by alansouzati** [comment]
Layer in mobile is broken

screen shot 2018-10-12 at 4 19 14 pm

**Comment by alansouzati** [comment]
Margin layer example is also broken

**Comment by alansouzati** [comment]
can we add a `row-responsive` in Box stories?

File: src/js/themes/base.js [hunk 1]
```diff
@@ -481,6 +493,7 @@ export const generate = (baseSpacing = 24, scale = 6) => { // 24
           large: { ...fontSizing(0) },
         },
       },
+      responsiveBreakpoint: 'small', // when we scale the font size down
```
[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/themes/base.js]: Yes. We would need to add `responsive={false}` to `Heading` to disable that behavior. Or, the theme could set `responsiveBreakpoint` to `undefined`.

**Comment by ericsoderberghp** [comment]
> can we add a `row-responsive` in Box stories?

SimpleBox is already `row-responsive`.

**Comment by ericsoderberghp** [comment]
> We need to update the `custom breakpoints` example. this pr breaks that. it would be good also because it will serve as a reference to folks wanting to leverage this.

Updated Custom Breakpoints story to align with theme changes.

**Comment by ericsoderberghp** [comment]
> Margin layer example is also broken

Layer stories should be working at small responsive sizes now.

**Comment by alansouzati** [comment]
This is definitely better!!

One last issue thought. For custom breakpoints example, you are passing:

```
      xsmall: {
        max: 499,
      },
      small: {
        min: 500,
        max: 899,
      },
      middle: {
        min: 900,
        max: 1535,
      },
```

So I thin we are in a conflicting situation here. There will be some overlap since we do a deep merge. I think as of now middle and medium overlap.

After adding that custom breakpoints, the font size is not changing anymore for the Heading. If I remove the custom breakpoints, the font size adjusts. So I believe this is an unexpected thing.

**Comment by alansouzati** [comment]
I think we should support this:

```
xsmall: {
        max: 499,
      },
      small: {
        min: 500,
        max: 899,
      },
      middle: {
        min: 900,
        max: 1535,
      },
      medium: undefined,
```

but as of now I get a null pointer expection

**Comment by ericsoderberghp** [comment]
I've fixed the overlap issue and the Heading issue.

**Comment by alansouzati** [comment]
tested everything looks good now!!

**Pull request merged**